src/main/java/com/conveyal/r5/profile/RaptorState.java

```diff
@@ -307,38 +287,30 @@ public String dump (int stop) {
      * will the downstream results change. This also clears the reached stops sets in preparation for a new round.
      */
     public void setDepartureTime(int departureTime) {
-        int previousDepartureTime = this.departureTime;
+        final int additionalWaitSeconds = this.departureTime - departureTime;
+        checkState(additionalWaitSeconds == 60, "Departure times may only be decremented by one minute.");
```

Comment:
👍

As we discussed, the departure time sampling for multi-criteria routing does not proceed minute-by-minute but that doesn't rely on range-raptor and is in the separate `McRaptorSuboptimalPathProfileRouter`.

Reply:
In that case I'd like to keep this constraint in place until we have some reason to change or remove it. I was originally only checking that the new departure time was earlier than the old, but tightened the check to reflect our actual practice and shrink the potential state space when debugging / reasoning about execution.